### PR TITLE
Added support for filtering javadocs

### PR DESCRIPTION
**Pull request by sahilasopa**

### CODE REVIEW

File: app/src/main/java/io/github/danthe1st/jdoc4droid/activities/list/classes/ListClassesActivity.java [hunk 1]
```diff
@@ -72,6 +76,31 @@ public void onCreate(Bundle savedInstanceState) {
...
+    public boolean onCreateOptionsMenu(Menu menu) {
+        boolean ret=super.onCreateOptionsMenu(menu);
+        filterButton=menu.findItem(R.id.app_bar_filter);
+        filterButton.setVisible(true);
+        filterButton.setOnMenuItemClickListener(menuItem -> {
+            filterButton.getSubMenu().clear();
+            for (String availableFilter : getAvailableFilters()) {
+                filterButton.getSubMenu().add(availableFilter);
+            }
+            filterButton.getSubMenu();
```
Comment (danthe1st) [inline app/src/main/java/io/github/danthe1st/jdoc4droid/activities/list/classes/ListClassesActivity.java]: What is the purpose of this line of code?

File: app/src/main/java/io/github/danthe1st/jdoc4droid/activities/list/classes/ListClassesActivity.java [hunk 2]
```diff
@@ -72,6 +76,31 @@ public void onCreate(Bundle savedInstanceState) {
...
+            return ret;
```
Comment (danthe1st) [inline app/src/main/java/io/github/danthe1st/jdoc4droid/activities/list/classes/ListClassesActivity.java]: This is unrelated to the return code of `onCreateOptionsMenu`. This line should be `return true;` in order to consume the event.

File: app/src/main/java/io/github/danthe1st/jdoc4droid/activities/list/classes/ListClassesActivity.java [hunk 3]
```diff
@@ -72,6 +76,31 @@ public void onCreate(Bundle savedInstanceState) {
...
     }
 
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        boolean ret=super.onCreateOptionsMenu(menu);
+        filterButton=menu.findItem(R.id.app_bar_filter);
+        filterButton.setVisible(true);
+        filterButton.setOnMenuItemClickListener(menuItem -> {
+            filterButton.getSubMenu().clear();
+            for (String availableFilter : getAvailableFilters()) {
```
Comment (danthe1st) [inline app/src/main/java/io/github/danthe1st/jdoc4droid/activities/list/classes/ListClassesActivity.java]: you can move the for-each-loop to another method (e.g. `loadFilters()`) and call it inside `runInUIThread()` inside `getThreadPool().execute()` in `onCreate` like the following:
```java
    @Override
    public void onCreate(Bundle savedInstanceState) {
        setContentView(R.layout.activity_list_classes_list);
        super.onCreate(savedInstanceState);
        javaDocDir = new File(getIntent().getStringExtra(ARG_JAVADOC_DIR));
        getThreadPool().execute(()-> {
            try {
                descriptions = JavaDocParser.loadClasses(javaDocDir);
                classTypes = descriptions.stream().map(SimpleClassDescription::getClassType).collect(Collectors.toSet());
                runInUIThread(()->{
                    adapter.setItems(descriptions);
                    findViewById(R.id.progressBar2).setVisibility(View.GONE);
                    reloadFilters();//call it here
                });
            } catch (Exception e) {
                runInUIThread(()->{
                    Toast.makeText(this,"Cannot load classes",Toast.LENGTH_LONG).show();
                    onBackPressed();
                    Log.e(getClass().getName(),"Cannot load class list",e);
                });
            }
        });
    }
```
```java
    @UiThread
    private void reloadFilters(){
        for (String availableFilter : getAvailableFilters()) {
            filterButton.getSubMenu().add(availableFilter);
        }
    }
```
This allows to remove the code clearing and re-filling the submenu every time it is clicked.

**Review by danthe1st — CHANGES_REQUESTED** [review]
Thank you for your pull request.

The text in the filter menu is invisible when using the light theme.

Please fix the mentioned problems so I can merge your PR.

File: app/src/main/java/io/github/danthe1st/jdoc4droid/activities/list/classes/ListClassesActivity.java (hunk 1 above)
Comment (sahilasopa) [reply app/src/main/java/io/github/danthe1st/jdoc4droid/activities/list/classes/ListClassesActivity.java]: this is  used to add items from Set<String> into menu items

File: app/src/main/java/io/github/danthe1st/jdoc4droid/activities/list/classes/ListClassesActivity.java (hunk 3 above)
Comment (sahilasopa) [reply app/src/main/java/io/github/danthe1st/jdoc4droid/activities/list/classes/ListClassesActivity.java]: alright makes sense

File: app/src/main/java/io/github/danthe1st/jdoc4droid/activities/list/classes/ListClassesActivity.java [hunk 4]
```diff
@@ -72,6 +80,23 @@ public void onCreate(Bundle savedInstanceState) {
...
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        boolean ret=super.onCreateOptionsMenu(menu);
+        filterButton=menu.findItem(R.id.app_bar_filter);
+        filterButton.setVisible(true);
+        return ret;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
```
Comment (danthe1st) [inline app/src/main/java/io/github/danthe1st/jdoc4droid/activities/list/classes/ListClassesActivity.java:92]: This methods is also called by other elements of the options menu.
Please run that code only when the selected item is the filer:
```java
    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        if(item.getItemId()==R.id.app_bar_filter){
            final String title = String.valueOf(item.getTitle());
            if (getAvailableFilters().contains(title)){
                onFilterChange(title);
            }
            return true;
        }else{
            return super.onOptionsItemSelected(item);
        }
    }
```

**Review by danthe1st — CHANGES_REQUESTED** [review]

**Review by danthe1st — APPROVED** [review]
